service/internal/builder/pipelines_builder.go

```diff
@@ -36,9 +36,9 @@ type builtPipeline struct {
 	firstMC consumer.Metrics
 	firstLC consumer.Logs
 
-	// MutatesConsumedData is set to true if any processors in the pipeline
+	// MutatesData is set to true if any processors in the pipeline
 	// can mutate the TraceData or MetricsData input argument.
-	MutatesConsumedData bool
+	MutatesData bool
 
 	processors []component.Processor
```

Comment:
Now that exporters can declare that they mutate data we need to using cloning fanout for them like we do for processors.